dexmaker/src/main/java/com/android/dx/DexMaker.java

```diff
@@ -479,27 +492,29 @@ ClassDefItem toClassDefItem() {
 
             CstType thisType = type.constant;
 
-            ClassDefItem out = new ClassDefItem(thisType, flags, supertype.constant,
-                    interfaces.ropTypes, new CstString(sourceFile));
-
-            for (MethodDeclaration method : methods.values()) {
-                EncodedMethod encoded = method.toEncodedMethod(dexOptions);
-                if (method.isDirect()) {
-                    out.addDirectMethod(encoded);
-                } else {
-                    out.addVirtualMethod(encoded);
+            if (null == classDefItem) {
```

Comment:
Hopefully we can remove the need for this check (see my other comment about state), but if not, let's be consistent and use `variable == null`

Reply:
NP. I will change to variable == null

Reply:
It's done by new commit 733f075